### PR TITLE
[Merged by Bors] - chore(NumberTheory/LucasPrimality): golf theorem

### PR DESCRIPTION
**Pull request by vihdzp**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/NumberTheory/LucasPrimality.lean [hunk 1]
```diff
@@ -39,23 +39,12 @@ group must itself have order `p-1`, which only happens when `p` is prime.
...
-  have hp' : p - 2 + 1 = p - 1 := tsub_add_eq_add_tsub hp1
-  let a' : (ZMod p)ˣ := Units.mkOfMulEqOne a (a ^ (p - 2)) (by rw [← pow_succ', hp', ha])
-  calc
-    p - 1 = orderOf a := order_of_a.symm
-    _ = orderOf a' := (orderOf_injective (Units.coeHom (ZMod p)) Units.ext a')
-    _ ≤ Fintype.card (ZMod p)ˣ := orderOf_le_card_univ
+  apply (Nat.card_units_zmod_lt_sub_one hp1).antisymm
+  let a' : (ZMod p)ˣ := Units.mkOfMulEqOne a _ (by rwa [← pow_succ', tsub_add_eq_add_tsub hp1])
+  rw [← orderOf_eq_of_pow_and_pow_div_prime (tsub_pos_of_lt hp1) ha hd]
+  exact (orderOf_injective (Units.coeHom _) Units.ext a').trans_le orderOf_le_card_univ
```
Comment (jcommelin) [inline Mathlib/NumberTheory/LucasPrimality.lean:50]: I liked the calc block. I would be happy to keep it. Though the `calc` keyword and the first line of the block could be merged onto one line.

File: Mathlib/NumberTheory/LucasPrimality.lean [hunk 2]
```diff
@@ -39,23 +39,12 @@ group must itself have order `p-1`, which only happens when `p` is prime.
...
-    exact hd 2 Nat.prime_two (dvd_zero _) (pow_zero _)
-  have h1 : p ≠ 1 := by
-    rintro ⟨⟩
-    exact hd 2 Nat.prime_two (dvd_zero _) (pow_zero _)
-  have hp1 : 1 < p := lt_of_le_of_ne h0.bot_lt h1.symm
-  have order_of_a : orderOf a = p - 1 := by
-    apply orderOf_eq_of_pow_and_pow_div_prime _ ha hd
-    exact tsub_pos_of_lt hp1
-  haveI : NeZero p := ⟨h0⟩
+  have h : p ≠ 0 ∧ p ≠ 1 := by
```
Comment (jcommelin) [inline Mathlib/NumberTheory/LucasPrimality.lean:42]: If you want, you could do
```suggestion
  obtain ⟨h0, h1⟩ : p ≠ 0 ∧ p ≠ 1 := by
```

Comment (vihdzp) [reply Mathlib/NumberTheory/LucasPrimality.lean:42]: I use the `p ≠ 0 ∧ p ≠ 1` hypothesis as is for `hp1`, so I think it's fine as is.

File: Mathlib/NumberTheory/LucasPrimality.lean (hunk 1 above)
Comment (vihdzp) [reply Mathlib/NumberTheory/LucasPrimality.lean]: Alright, I'll revert that change.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**